File: Daily challenge/Distribute Coins in Binary Tree.cpp
class Solution 
{
public:
    int res = 0;

    pair<int , int> DFS(TreeNode* node)
    {
        if(!node)
        {
            return {0, 0};
        }

        pair<int, int> left = DFS(node->left);
        pair<int, int> right = DFS(node->right);

        int size = 1 + left.first + right.first;
        int coins = node->val + left.second + right.second;
        res += abs(size - coins);
        return {size, coins};
    }
    int distributeCoins(TreeNode* root) 
    {
        DFS(root);
        return res;
    }
};

File: Daily challenge/Distribute Coins in Binary Tree.py
class Solution:
    res = 0
    def DFS(self, node):
        if not node:
            return [0, 0]
        
        L_size, L_coins = self.DFS(node.left)
        R_size, R_coins = self.DFS(node.right)

        size = 1 + L_size + R_size
        coins = node.val + L_coins + R_coins
        self.res += abs(size - coins)

        return [size, coins]
        

    def distributeCoins(self, root: Optional[TreeNode]) -> int:
        self.DFS(root)
        return self.res